src/app.py

```diff
@@ -4212,6 +4213,155 @@ def multiple_components():
+
+    Returns
+    -------
+    dict
+        The results of the bulk update. The key is the uuid of the entity. If
+        successful, the value is a dictionary with "success" as True and "data" as the
+        entity data. If failed, the value is a dictionary with "success" as False and
+        "data" as the error message.
+    """
+    headers = {
+        "Authorization": f"Bearer {token}",
+        "X-Application": "entity-api",
+    }
+    # create a session with retries
+    session = requests.Session()
+    session.headers = headers
+    retries = Retry(
+        total=total_tries,
+        backoff_factor=throttle,
+        status_forcelist=[500, 502, 503, 504],
+    )
+    session.mount(entity_api_url, HTTPAdapter(max_retries=retries))
+
+    results = {}
+    with session as s:
+        for idx, (uuid, payload) in enumerate(entity_updates.items()):
+            try:
+                # https://github.com/hubmapconsortium/entity-api/issues/698#issuecomment-2260799700
+                # yuanzhou: When you iterate over the target uuids make individual PUT /entities/<uuid> calls.
+                # The main reason we use the PUT call rather than direct neo4j query is because the entity update
+                # needs to go through the schema trigger methods and generate corresponding values programmatically
+                # before sending over to neo4j.
+                # The PUT call returns the response immediately while the backend updating may be still going on.
+                res = s.put(
+                    f"{entity_api_url}/entities/{uuid}", json=payload, timeout=15
+                )
+                results[uuid] = {
+                    "success": res.ok,
+                    "data": res.json() if res.ok else res.json().get("error"),
+                }
+            except requests.exceptions.RequestException as e:
+                logger.error(f"Failed to update entity {uuid}: {e}")
+                results[uuid] = {"success": False, "data": str(e)}
+
+            if after_each_callback:
+                after_each_callback(idx)
+
+            if idx < len(entity_updates) - 1:
+                time.sleep(throttle)
+
+    return results
+
+
+def update_datasets_uploads(entity_updates: list, token: str, entity_api_url: str) -> None:
+    """
+    For this call to work READ_ONLY_MODE = False in the app.cfg file.
+    """
+    update_payload = {ds.pop("uuid"): ds for ds in entity_updates}
+
+    # send the dataset/upload updates to entity-api
+    update_res = bulk_update_entities(update_payload, token, entity_api_url)
+
+    for uuid, res in update_res.items():
+        if not res["success"]:
+            logger.error(f"Failed to update entity {uuid}: {res['data']}")
+
+
+ENTITY_BULK_UPDATE_FIELDS_ACCEPTED = ['uuid', 'status', 'ingest_task', 'assigned_to_group_name']
+
+
+@app.route('/datasets', methods=['PUT'])
+@app.route('/uploads', methods=['PUT'])
+def entity_bulk_update():
```

Comment:
Please also add function comment block to describe this is used by Data Ingest Board application for now.